llvm/lib/Target/RISCV/RISCVISelDAGToDAG.cpp

```diff
@@ -3829,8 +3830,9 @@ bool RISCVDAGToDAGISel::doPeepholeMaskedRVV(MachineSDNode *N) {
 #endif
 
   SmallVector<SDValue, 8> Ops;
-  // Skip the passthru operand at index 0 if !UseTUPseudo and no GPR out.
-  bool ShouldSkip = !UseTUPseudo && !hasGPROut(Opc);
+  // Skip the passthru operand at index 0 if !UseTUPseudo and has the passthru
+  // operand.
+  bool ShouldSkip = !UseTUPseudo && hasPassthru(MCID);
```

Comment:
I thought it was weird that we were using the vec policy op to determine if the unmasked pseudo had a passthru operand, so I fixed it to use `isFirstDefTiedToFirstUse` and remove `hasGPROut` in ec5d17b58779e2fb08a7c9c706b2842586f71f78, can you rebase on top of it and check if it works for stores now?

Reply:
Yes it works. And I agree it's cleaner in this way, thanks.